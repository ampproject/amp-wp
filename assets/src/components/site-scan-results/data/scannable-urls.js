export default [
	{
		url: 'https://example.org/',
		type: 'is_front_page',
		label: 'Homepage',
		amp_url: 'https://example.org/?amp=1',
		validation_errors: [
			{
				code: 'CSS_SYNTAX_INVALID_PROPERTY_NOLIST',
				css_property_name: 'behavior',
				css_property_value: "url('/bad.htc')",
				type: 'css_error',
				spec_name: 'style amp-custom',
				node_name: 'link',
				parent_name: 'head',
				node_attributes: {
					rel: 'stylesheet',
					id: 'bad-block-front-block-style-handle-css',
					href: 'https://example.org/wp-content/plugins/bad-block/front-block-style-handle.css?ver=__normalized__',
					media: 'all',
				},
				node_type: 1,
				sources: [
					{
						type: 'plugin',
						name: 'bad-block',
						file: 'bad-block.php',
						line: 42,
						function: 'Bad\\Block\\bad_block_init',
						hook: 'init',
						priority: 10,
						dependency_type: 'style',
						handle: 'bad-block-front-block-style-handle',
					},
					{
						type: 'core',
						name: 'wp-includes',
						file: 'script-loader.php',
						line: 2223,
						function:
							'wp_enqueue_registered_block_scripts_and_styles',
						hook: 'enqueue_block_assets',
						priority: 10,
						dependency_type: 'style',
						handle: 'bad-block-front-block-style-handle',
					},
				],
			},
			{
				node_name: 'script',
				parent_name: 'head',
				code: 'DISALLOWED_TAG',
				type: 'js_error',
				node_attributes: {
					src: 'https://example.org/wp-content/plugins/bad-block/front-block-script-handle.js?ver=__normalized__',
					id: 'bad-block-front-block-script-handle-js',
				},
				node_type: 1,
				sources: [
					{
						type: 'core',
						name: 'wp-includes',
						file: 'script-loader.php',
						line: 1952,
						function: 'wp_print_head_scripts',
						hook: 'wp_head',
						priority: 9,
					},
				],
			},
		],
		validated_url_post: {
			id: 15350,
			edit_link:
				'https://example.org/wp-admin/post.php?post=15350&action=edit',
		},
		stale: false,
	},
];
