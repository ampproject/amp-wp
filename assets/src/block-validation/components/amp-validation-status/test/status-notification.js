/**
 * External dependencies
 */
import { act } from 'react-dom/test-utils';

/**
 * WordPress dependencies
 */
import { render, unmountComponentAtNode } from '@wordpress/element';
import { useDispatch, useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import AMPValidationStatusNotification from '../status-notification';

jest.mock('@wordpress/data/build/components/use-select', () => jest.fn());
jest.mock('@wordpress/data/build/components/use-dispatch/use-dispatch', () =>
	jest.fn()
);

describe('AMPValidationStatusNotification', () => {
	let container;

	const autosave = jest.fn();
	const savePost = jest.fn();

	function setupUseSelect(overrides) {
		useSelect.mockImplementation(() => ({
			fetchingErrorsRequestErrorMessage: '',
			isDraft: false,
			isEditedPostNew: false,
			isFetchingErrors: false,
			keptMarkupValidationErrorCount: 0,
			reviewLink: 'http://example.com',
			supportLink: 'http://example.com/support',
			unreviewedValidationErrorCount: 0,
			validationErrorCount: 0,
			...overrides,
		}));
	}

	beforeAll(() => {
		useDispatch.mockImplementation(() => ({ autosave, savePost }));
	});

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		unmountComponentAtNode(container);
		container.remove();
		container = null;
	});

	it('does not render when errors are being fetched', () => {
		setupUseSelect({
			isFetchingErrors: true,
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.children).toHaveLength(0);
	});

	it('renders message when the post is new', () => {
		setupUseSelect({
			isEditedPostNew: true,
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain(
			'Validation will be checked upon saving'
		);
		expect(
			container.querySelector('a[href="http://example.com"]')
		).toBeNull();
	});

	it('renders error message when API request error is present', () => {
		setupUseSelect({
			fetchingErrorsRequestErrorMessage: 'request error message',
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain('request error message');
		expect(
			container.querySelector('a[href="http://example.com"]')
		).toBeNull();

		container.querySelector('button').click();
		expect(autosave).toHaveBeenCalledWith({ isPreview: true });

		setupUseSelect({
			isDraft: true,
			fetchingErrorsRequestErrorMessage: 'request error message',
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});
		container.querySelector('button').click();
		expect(savePost).toHaveBeenCalledWith({ isPreview: true });
	});

	it('renders error message when there are kept issues', () => {
		setupUseSelect({
			keptMarkupValidationErrorCount: 2,
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain(
			'AMP is disabled due to invalid markup being kept for 2 issues.'
		);
		expect(
			container.querySelector('a[href="http://example.com"]')
		).not.toBeNull();
	});

	it('renders message when there are unreviewed issues', () => {
		setupUseSelect({
			unreviewedValidationErrorCount: 3,
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain(
			'AMP is valid, but 3 issues need review.'
		);
		expect(
			container.querySelector('a[href="http://example.com"]')
		).not.toBeNull();
	});

	it('renders message when there are reviewed issues', () => {
		setupUseSelect({
			validationErrorCount: 1,
		});

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain(
			'AMP is valid. 1 issue was reviewed.'
		);
		expect(
			container.querySelector('a[href="http://example.com"]')
		).not.toBeNull();
	});

	it('renders message when there are no AMP validation errors', () => {
		setupUseSelect();

		act(() => {
			render(<AMPValidationStatusNotification />, container);
		});

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.innerHTML).toContain(
			'No AMP validation issues detected.'
		);
	});
});
