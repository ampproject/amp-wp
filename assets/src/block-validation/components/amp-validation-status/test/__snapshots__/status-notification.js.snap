// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMPValidationStatusNotification renders error message when API request error is present 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><svg width=\\"21\\" height=\\"21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g clip-path=\\"url(#clip-amp-validation-errors-kept)\\" fill=\\"#BB522E\\"><path d=\\"M10.762 2.541c4.4 0 8 3.6 8 8 0 1.6-.5 3-1.2 4.2l1.4 1.5c1.1-1.6 1.8-3.6 1.8-5.7 0-5.5-4.5-10-10-10-2 0-3.9.6-5.5 1.7l1.4 1.5c1.2-.8 2.6-1.2 4.1-1.2ZM.762 10.541c0 5.5 4.5 10 10 10 2.7 0 5.1-1.1 6.9-2.8l-14-14.2c-1.8 1.8-2.9 4.3-2.9 7Zm10 8c-4.4 0-8-3.6-8-8 0-1.5.4-2.8 1.1-4l10.9 10.9c-1.2.7-2.5 1.1-4 1.1Z\\"></path><path d=\\"M14.262 9.74c.1 0 .1-.1.1-.2 0-.2-.2-.4-.4-.4h-2l1.6 1.6.7-1ZM12.461 4.541h-.8l-1.6 2.6 1.7 1.7.7-4.3ZM7.462 11.541s-.1.1-.1.2c0 .2.2.4.4.4h2.3l-.8 4.5h.7l2.6-4.1-3.5-3.6-1.6 2.6Z\\"></path></g><defs><clipPath id=\\"clip-amp-validation-errors-kept\\"><path fill=\\"#fff\\" transform=\\"translate(.762 .541)\\" d=\\"M0 0h20v20H0z\\"></path></clipPath></defs></svg></div><div class=\\"sidebar-notification__content\\"><p>request error message</p><div class=\\"sidebar-notification__action\\"><button type=\\"button\\" class=\\"components-button is-link\\">Try again</button></div></div></div>"`;

exports[`AMPValidationStatusNotification renders error message when there are kept issues 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><svg width=\\"21\\" height=\\"21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g clip-path=\\"url(#clip-amp-validation-errors-kept)\\" fill=\\"#BB522E\\"><path d=\\"M10.762 2.541c4.4 0 8 3.6 8 8 0 1.6-.5 3-1.2 4.2l1.4 1.5c1.1-1.6 1.8-3.6 1.8-5.7 0-5.5-4.5-10-10-10-2 0-3.9.6-5.5 1.7l1.4 1.5c1.2-.8 2.6-1.2 4.1-1.2ZM.762 10.541c0 5.5 4.5 10 10 10 2.7 0 5.1-1.1 6.9-2.8l-14-14.2c-1.8 1.8-2.9 4.3-2.9 7Zm10 8c-4.4 0-8-3.6-8-8 0-1.5.4-2.8 1.1-4l10.9 10.9c-1.2.7-2.5 1.1-4 1.1Z\\"></path><path d=\\"M14.262 9.74c.1 0 .1-.1.1-.2 0-.2-.2-.4-.4-.4h-2l1.6 1.6.7-1ZM12.461 4.541h-.8l-1.6 2.6 1.7 1.7.7-4.3ZM7.462 11.541s-.1.1-.1.2c0 .2.2.4.4.4h2.3l-.8 4.5h.7l2.6-4.1-3.5-3.6-1.6 2.6Z\\"></path></g><defs><clipPath id=\\"clip-amp-validation-errors-kept\\"><path fill=\\"#fff\\" transform=\\"translate(.762 .541)\\" d=\\"M0 0h20v20H0z\\"></path></clipPath></defs></svg></div><div class=\\"sidebar-notification__content\\"><p>AMP is disabled due to invalid markup being kept for 2 issues.</p><div class=\\"sidebar-notification__action\\"><a class=\\"components-external-link\\" href=\\"http://example.com\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">View technical details<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a><br><a class=\\"components-external-link\\" href=\\"http://example.com/support\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">Get Support<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a></div></div></div>"`;

exports[`AMPValidationStatusNotification renders message when the post is new 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"m13.3 9.1-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>Validation will be checked upon saving.</p></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are no AMP validation errors 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"m13.3 9.1-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>No AMP validation issues detected.</p></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are reviewed issues 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"m13.3 9.1-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>AMP is valid. 1 issue was reviewed.</p><div class=\\"sidebar-notification__action\\"><a class=\\"components-external-link\\" href=\\"http://example.com\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">View technical details<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a><br><a class=\\"components-external-link\\" href=\\"http://example.com/support\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">Get Support<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a></div></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are unreviewed issues 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon amp-status-icon--broken\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"m13.3 9.1-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>AMP is valid, but 3 issues need review.</p><div class=\\"sidebar-notification__action\\"><a class=\\"components-external-link\\" href=\\"http://example.com\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">View technical details<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a><br><a class=\\"components-external-link\\" href=\\"http://example.com/support\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">Get Support<span data-wp-c16t=\\"true\\" data-wp-component=\\"VisuallyHidden\\" class=\\"components-visually-hidden css-1mm2cvy-View em57xhy0\\" style=\\"border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-16iaek2-StyledIcon esh4a730\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a></div></div></div>"`;
